e2e/fiat_tf_test.go

```diff
@@ -0,0 +1,209 @@
+}
+
+func TestFiatTFPause(t *testing.T) {
+	if testing.Short() {
+		t.Skip()
+	}
+	t.Parallel()
+
+	ctx := context.Background()
+
+	nw := nobleSpinUp(t, ctx, true)
+	noble := nw.chain
+	val := noble.Validators[0]
+
+	// ACTION: Pause TF from an account that is not the Pauser
+	// EXPECTED: Request fails; TF not paused
+
+	w := interchaintest.GetAndFundTestUsers(t, ctx, "alice", math.OneInt(), noble)
+	alice := w[0]
+
+	_, err := val.ExecTx(ctx, alice.KeyName(), "fiat-tokenfactory", "pause")
+	require.ErrorContains(t, err, "you are not the pauser: unauthorized")
+
+	showPausedRes, err := showPaused(ctx, val)
+	require.NoError(t, err, "error querying for paused state")
+	expectedPaused := fiattokenfactorytypes.QueryGetPausedResponse{
+		Paused: fiattokenfactorytypes.Paused{
+			Paused: false,
+		},
+	}
+	require.Equal(t, expectedPaused, showPausedRes)
+
+	// ACTION: Pause TF from a blacklisted Pauser account
+	// EXPECTED: Success; TF is paused
+	// Status:
+	// 	Paused: false
+
+	blacklistAccount(t, ctx, val, nw.fiatTfRoles.Blacklister, nw.fiatTfRoles.Pauser)
+
+	pauseFiatTF(t, ctx, val, nw.fiatTfRoles.Pauser)
+
+	unblacklistAccount(t, ctx, val, nw.fiatTfRoles.Blacklister, nw.fiatTfRoles.Pauser)
+
+	// ACTION: Pause TF while TF is already paused
+	// EXPECTED: Success; TF remains paused
+	// Status:
+	// 	Paused: true
+
+	pauseFiatTF(t, ctx, val, nw.fiatTfRoles.Pauser)
+}
+
+func TestFiatTFUnpause(t *testing.T) {
+	if testing.Short() {
+		t.Skip()
+	}
+	t.Parallel()
+
+	ctx := context.Background()
+
+	nw := nobleSpinUp(t, ctx, true)
+	noble := nw.chain
+	val := noble.Validators[0]
+
+	// ACTION: Unpause TF from an account that is not a Pauser
+	// EXPECTED: Request fails; TF remains paused
+
+	w := interchaintest.GetAndFundTestUsers(t, ctx, "alice", math.OneInt(), noble)
+	alice := w[0]
+
+	pauseFiatTF(t, ctx, val, nw.fiatTfRoles.Pauser)
+
+	_, err := val.ExecTx(ctx, alice.KeyName(), "fiat-tokenfactory", "unpause")
+	require.ErrorContains(t, err, "you are not the pauser: unauthorized")
+
+	showPausedRes, err := showPaused(ctx, val)
+	require.NoError(t, err, "error querying for paused state")
+	expectedPaused := fiattokenfactorytypes.QueryGetPausedResponse{
+		Paused: fiattokenfactorytypes.Paused{
+			Paused: true,
+		},
+	}
+	require.Equal(t, expectedPaused, showPausedRes)
+
+	// ACTION: Unpause TF from a blacklisted Pauser account
+	// EXPECTED: Success; TF is unpaused
+	// Status:
+	// 	Paused: true
+
+	blacklistAccount(t, ctx, val, nw.fiatTfRoles.Blacklister, nw.fiatTfRoles.Pauser)
+
+	unpauseFiatTF(t, ctx, val, nw.fiatTfRoles.Pauser)
+
+	unblacklistAccount(t, ctx, val, nw.fiatTfRoles.Blacklister, nw.fiatTfRoles.Pauser)
+
+	// ACTION: Unpause TF while TF is already paused
+	// EXPECTED: Success; TF remains unpaused
+	// Status:
+	// 	Paused: false
+
+	pauseFiatTF(t, ctx, val, nw.fiatTfRoles.Pauser)
+
+	unpauseFiatTF(t, ctx, val, nw.fiatTfRoles.Pauser)
+
+	unpauseFiatTF(t, ctx, val, nw.fiatTfRoles.Pauser)
+}
```

Comment:
Can we get a happy path test here as well? So unpausing a paused TF with an unblacklisted pauser.

Reply:
Added in: [`9a77cc4` (#369)](https://github.com/noble-assets/noble/pull/369/commits/9a77cc480b7a8b9d0534414f76a3f8325109702d)